support/k8s/k8s-workload-registrar/webhookhandler.go

```diff
@@ -3,25 +3,104 @@ package main
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/klog/v2"
 	"net/http"
 
+	admv1 "k8s.io/api/admission/v1"
 	admv1beta1 "k8s.io/api/admission/v1beta1"
 )
 
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	deserializer  = codecs.UniversalDeserializer()
+)
+
 type AdmissionController interface {
-	ReviewAdmission(context.Context, *admv1beta1.AdmissionRequest) (*admv1beta1.AdmissionResponse, error)
+	ReviewAdmission(context.Context, admv1.AdmissionReview) (*admv1.AdmissionResponse, error)
 }
 
 type WebhookHandler struct {
 	controller AdmissionController
 }
 
 func NewWebhookHandler(controller AdmissionController) *WebhookHandler {
+	_ = admv1.AddToScheme(runtimeScheme)
 	return &WebhookHandler{
 		controller: controller,
 	}
 }
 
+// admitv1beta1Func handles a v1beta1 admission
+type admitv1beta1Func func(context.Context, admv1beta1.AdmissionReview) (*admv1beta1.AdmissionResponse, error)
+
+// admitv1beta1Func handles a v1 admission
+type admitv1Func func(context.Context, admv1.AdmissionReview) (*admv1.AdmissionResponse, error)
+
+// admitHandler is a handler, for both validators and mutators, that supports multiple admission review versions
+type admitHandler struct {
+	v1beta1 admitv1beta1Func
+	v1      admitv1Func
+}
+
+func newDelegateToV1AdmitHandler(f admitv1Func) admitHandler {
+	return admitHandler{
+		v1beta1: delegateV1beta1AdmitToV1(f),
+		v1:      f,
+	}
+}
+
+func delegateV1beta1AdmitToV1(f admitv1Func) admitv1beta1Func {
+	return func(context context.Context, review admv1beta1.AdmissionReview) (*admv1beta1.AdmissionResponse, error) {
+		in := admv1.AdmissionReview{Request: convertAdmissionRequestToV1(review.Request)}
+		out, err := f(context, in)
+		if err != nil {
+			return nil, err
+		}
+		return convertAdmissionResponseToV1beta1(out), nil
+	}
+}
+
+func convertAdmissionRequestToV1(r *admv1beta1.AdmissionRequest) *admv1.AdmissionRequest {
+	return &admv1.AdmissionRequest{
+		Kind:               r.Kind,
+		Namespace:          r.Namespace,
+		Name:               r.Name,
+		Object:             r.Object,
+		Resource:           r.Resource,
+		Operation:          admv1.Operation(r.Operation),
+		UID:                r.UID,
+		DryRun:             r.DryRun,
+		OldObject:          r.OldObject,
+		Options:            r.Options,
+		RequestKind:        r.RequestKind,
+		RequestResource:    r.RequestResource,
+		RequestSubResource: r.RequestSubResource,
+		SubResource:        r.SubResource,
+		UserInfo:           r.UserInfo,
+	}
+}
+
+func convertAdmissionResponseToV1beta1(r *admv1.AdmissionResponse) *admv1beta1.AdmissionResponse {
+	var pt *admv1beta1.PatchType
+	if r.PatchType != nil {
+		t := admv1beta1.PatchType(*r.PatchType)
+		pt = &t
+	}
+	return &admv1beta1.AdmissionResponse{
+		UID:              r.UID,
+		Allowed:          r.Allowed,
+		AuditAnnotations: r.AuditAnnotations,
+		Patch:            r.Patch,
+		PatchType:        pt,
+		Result:           r.Result,
```

Comment:
Do we need `Warnings:         r.Warnings,`?

Reply:
now, there is no such `Warnings` field in the struct. 🙋🏻‍♂️🤝